src/core/Flyout.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useState } from "react";
+import {
+  NavigationMenu,
+  NavigationMenuItem,
+  NavigationMenuList,
+  NavigationMenuTrigger,
+  NavigationMenuContent,
+  NavigationMenuViewport,
+  NavigationMenuLink,
+} from "@radix-ui/react-navigation-menu";
+import cn from "./utils/cn";
+import { componentMaxHeight, HEADER_HEIGHT } from "./utils/heights";
+
+/**
+ * Props for the Flyout component.
+ */
+type FlyoutProps = {
+  /**
+   * Array of menu items to be displayed in the flyout.
+   */
+  menuItems: {
+    /**
+     * Label for the menu item.
+     */
+    label: string;
+    /**
+     * Optional content to be displayed in the flyout panel.
+     */
+    content?: React.ReactNode;
+    /**
+     * Optional link for the menu item.
+     */
+    link?: string;
+    /**
+     * Optional styling for the flyout panel.
+     */
+    panelStyling?: string;
```

Comment:
This contradicts the `...ClassName` conventions used elsewhere

Reply:
thanks missed this, will update